pkg/apis/numaflow/v1alpha1/mono_vertex_types.go

```diff
@@ -529,14 +529,21 @@ func (mvs *MonoVertexStatus) MarkPhaseRunning() {
 }
 
 // IsHealthy indicates whether the MonoVertex is in healthy status
-func (mvs *MonoVertexStatus) IsHealthy() bool {
+// It returns an error if any issues exists
+// No errors indicate that the MonoVertex is healthy
+// TODO: Add support for paused whenever added in MonoVtx?
+func (mvs *MonoVertexStatus) IsHealthy() error {
```

Comment:
Instead changing this function, use the `reason` and `message` for unhealthy message.

Reply:
To extract the `reason` and `message` from the conditions I had updated the message. Do you prefer extracting that directly from the new function `mvs.IsReadyWithError()` instead for that? And not passing it through this one

Reply:
Okay, Ignore the above understood what you meant. Looked into the spec.

I had looked into healthy status message, thus reason and message were not present hence I didn't realise that we would set those fields are also present when there is an error. 

Will update the code. 

```
status:
  conditions:
  - lastTransitionTime: "2024-08-13T17:09:16Z"
    message: Successful
    reason: Successful
    status: "True"
    type: DaemonHealthy
  - lastTransitionTime: "2024-08-13T17:09:10Z"
    message: Successful
    reason: Successful
    status: "True"
    type: Deployed
  - lastTransitionTime: "2024-08-13T17:09:10Z"
    message: All vertex pods are healthy
    reason: Running
    status: "True"
    type: PodsHealthy
  lastScaledAt: "2024-08-13T17:09:10Z"
  lastUpdated: null
  observedGeneration: 1
  phase: Running
  replicas: 1
  selector: app.kubernetes.io/component=mono-vertex,numaflow.numaproj.io/mono-vertex-name=simple-mono-vertex
```